### PR TITLE
Delegate socket protect callback to tunnel setup

### PR DESCRIPTION
**Pull request by tadovas**
This one is important for mobile node

### CODE REVIEW

File: openvpn3/tun_setup.go [hunk 1]
```diff
@@ -166,4 +167,8 @@ func (setup *NoOpTunnelSetup) Teardown(disconnect bool) {
 
 }
 
+func (setup *NoOpTunnelSetup) SocketProtect(socket int) bool {
```
Comment (soffokl) [inline openvpn3/tun_setup.go]: CI failed because this method does not have a proper comment.

File: openvpn3/tun_setup_bridge.go [hunk 2]
```diff
@@ -272,6 +274,14 @@ func goTeardown(cd C.user_callback_data, disconnect C.bool) {
...
 }
 
+//export goSocketProtect
+func goSocketProtect(cd C.user_callback_data, socket C.int) C.bool {
+	id := int(cd)
+	delegate := tunnelSetupRegistry.lookup(id)
+	return C.bool(delegate.SocketProtect(int(socket)))
+}
+
+
```
Comment (soffokl) [inline openvpn3/tun_setup_bridge.go]: 2 empty lines are not allowed for `gofmt`. CI failed because of it.

File: openvpn3/tun_setup_bridge.go [hunk 3]
```diff
@@ -306,5 +316,6 @@ func registerTunnelSetupDelegate(delegate TunnelSetup) (C.tun_builder_callbacks,
 		persist:                   C.tun_builder_persist(C.goPersist),
 		establish_lite:            C.tun_builder_establish_lite(C.goEstablishLite),
 		teardown:                  C.tun_builder_teardown(C.goTeardown),
+		socket_protect:  		   C.tun_socket_protect(C.goSocketProtect),
```
Comment (soffokl) [inline openvpn3/tun_setup_bridge.go]: Not formatted with `gofmt`.

**Comment by tadovas** [comment]
@soffokl nice catches. Fixed Ci complains.

**Pull request merged**